src/util/address.rs

```diff
@@ -168,22 +337,19 @@ impl Payload {
     /// Get a [Payload] from an output script (scriptPubkey).
     pub fn from_script(script: &script::Script) -> Option<Payload> {
         Some(if script.is_p2pkh() {
-            Payload::PubkeyHash(PubkeyHash::from_slice(&script.as_bytes()[3..23]).unwrap())
+            Payload::PubkeyHash(PubkeyHash::from_slice(&script.as_bytes()[3..23]).expect("hash engine broken"))
```

Comment:
The failure has nothing to do with the hash engine but with supplying a byte slice of a wrong length.

Reply:
Removed `expect` at all by using infallible `from_inner`